internal/k6runner/k6runner.go

```diff
@@ -33,6 +41,31 @@ type Settings struct {
 	Timeout int64 `json:"timeout"`
 }
 
+// CheckInfo holds information about the SM check that triggered this script.
+type CheckInfo struct {
+	// Type is the string representation of the check type this script belongs to (browser, scripted, multihttp, etc.)
+	Type string `json:"type"`
+	// Metadata is a collection of key/value pairs containing information about this check, such as check and tenant ID.
+	// It is loosely typed on purpose: Metadata should only be used for informational properties that will make its way
+	// into telemetry, and not for making decision on it.
+	Metadata map[string]any `json:"metadata"`
+}
```

Comment:
This is where I'd like to have more reviewer attention, as these changes are hard to revert.

I chose to separate actionable information (`Type`), which will be used downstream to make decisions, from non-actionable data that will be basically used to enrich logging. I think this makes the relevant information "strongly typed", if that makes sense. Looking forward to your thoughts on this.

Reply:
> That's kind of the reason for using map[string]string instead of map[string]any.

I'm fine-ish with `map[string]string` for metadata. I leaned towards `any` because that will show integers as integers in the runner logs, instead of strings. If at some point we want to correlate logs, I think it would be nice that what is logged as an int in one place is also logged as an int on another.

> Unless you introduce actual coupling between the agent and the runner, which is not going to happen, this is all in internal/, the "strong type" argument is not very strong. I'm not saying it's wrong, just that it's not giving you what you say it gives you.

Practically it's giving nothing, but I think the separation conveys meaning to readers. They will notice that particular field is special and think twice before adding or removing it.

Reply:
OK, I'm good with that, I just wanted to make sure the expectations are correct.

Do note that JSON numbers are super weird, because the JS in JSON stands for JavaScript, and JavaScript has no integers, only floats.

JSON mixed with Go is even weirder, because, when dealing with `any`, Go says "if JSON numbers are floats then I will decode them into float64 because that is what makes sense". You can override that using [UseNumber](https://pkg.go.dev/encoding/json#Decoder.UseNumber) (which requires using a `json.Decoder` not just `json.Unmarshal`) plus asking nicely for the specific conversion you want.

Do note this has an implication when marshaling elements from `map[string]any` into JSON via zerolog: the JSON decoder placed a float64 in the map's value, so `logger.Any(...)` sees a float, and it will log it out as a float. If you go thru the `json.Number` route, I think that should do the right thing (because `json.Number`'s underlying type is `string` 😆).